homework/index.js

```diff
@@ -1,46 +1,180 @@
+// Add new functions and modify function main() as you see fit.
+
+function main(url) {
+  fetchJSON(url, (err, repositories) => {
+    if (err) {
+      createAndAppend('div', root, {
+        text: err.message,
+        class: 'alert-error',
+      });
+      return;
+    }
+    // ref: console.log(repositories);
+
+    //SORT repo names in alphabetical order (case-insensitive) & generate obj with REDUCE
+
+    /* REF https://stackoverflow.com/questions/43773092/how-to-sort-objects-by-value
+        let obj = {a: {'name':'Boy'}, b:{'name': 'apple'}, c:{'name':'Dog'}, d:{'name':'cat'}};
+        Object.keys(obj).sort((a, b) => {
+            a = obj[a].name.toLowerCase();
+            b = obj[b].name.toLowerCase();
+            if (a == b) return 0;
+            return a < b ? -1 : 1;
+        })
+        //return ["b", "a", "d", "c"]
+        */
+    let sorted = Object.keys(repositories)
+      .sort((a, b) => {
+        a = repositories[a].name.toLowerCase();
+        b = repositories[b].name.toLowerCase();
+        if (a === b) return 0;
+        return a < b ? -1 : 1;
+      })
+      .reduce((acc, curr, i) => {
+        acc[i] = repositories[curr];
+        return acc;
+      }, {});
+
+    // ref: console.log(sorted);
+
+    // generate ARRAY of repo names
+    for (const repo in sorted) {
+      repoNamesArr.push(sorted[repo].name);
+    }
+
+    // ref:   console.log(repoNamesArr);
+
+    // create list of repo names from Arr to select from
+    repoNamesArr.forEach(repoName => {
+      createAndAppend('option', select, {
+        text: repoName,
+        value: repoNamesArr.indexOf(repoName), // where each option element has the array index of the repository as its value
+      });
     });
-  }
 
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
-  window.onload = () => main(HYF_REPOS_URL);
+    // WHEN SELECTING A REPO ...
+    select.addEventListener('change', repo => {
+      // SHOW repo info
+      selected.innerHTML = '';
+
+      /* ref 
+      console.log(repo);
+      console.log(repo.target);
+      console.log(repo.target.value);
+      console.log(sorted[repo.target.value].url);
+      */
+
+      createAndAppend('div', selected, {
+        text: `<b>Name of Repository:</b><br><a href="${
+          sorted[repo.target.value].html_url
+        }" target="_blank">${sorted[repo.target.value].name}</a><br><br>`,
+        class: 'selected',
+      });
+      createAndAppend('div', selected, {
+        text: `<b>Description:</b><br>${sorted[repo.target.value].description}<br><br>`,
+        class: 'selected',
+      });
+      createAndAppend('div', selected, {
+        text: `<b>Fork(s)</b>:<br>${sorted[repo.target.value].forks}<br><br>`,
+        class: 'selected',
+      });
+      createAndAppend('div', selected, {
+        text: `<b>Updated at</b>:<br>${sorted[repo.target.value].updated_at}<br><br>`,
+        class: 'selected',
+      });
```

Comment:
Ah, I see that the `createAndAppend` helper function doesn't really cover the case of trying to set innerHTML. Looks like you got it to work, but maybe the helper function needs changed 🤔